### ISSUE TITLE
Can not set temperature

### ISSUE THREAD

**Issue by Meewis** [dehsgr/node-red-contrib-evohome#19]
Using the last version.

I can read the setting of my thermostat:
{"id":"7646762","name":"thermostat","currentTemperature":19,"targetTemperature":11.5,"systemModeStatus":{"mode":"Auto","isPermanent":true}}

But when I want to sent a new setting nothing seems to happen. Also no error messages on the cli
"{ id: "7646762",  temperature: 15, permanent: true }"

When send a new setting by the Honywell app it works.

Anything I did wrong?
Please help.
Kind regards, Wilco

**Comment by dehsgr** [comment]
Hey Wilco,

can you try a rollback to the previous version to confirm that version is working as expected?

**Comment by Meewis** [comment]
Hi dehsgr,

I just updated from 1.1.2 to 1.2.0 because of this consisting problem. On both no success. To update to 1.2.0 I had to replace the map at node_modules, because with the node-red "manage pallet" the new version is not (jet) available.

**Comment by dehsgr** [comment]
@Meewis so this isn't a new issue from 1.2.0. Am I right? The updated node show be recognized from palette shortly.

**Comment by Meewis** [comment]
Indeed, I'm struggling already some months. And sometimes the message worked, but mostly not.

**Comment by dehsgr** [comment]
Sorry. It is very difficult for me to help you, because I don't have any evohome to test here. I developed this node for a friend of mine. Maybe this issue could be due to sending to much commands to your evohome. Try to decrease amount of commands.

**Comment by Meewis** [comment]
OK, I do understand. Then maybe you can help me sorting it out; on the first place I'm watching the node-red context data and there is the global var "evohome-session" visible, but the content is "[object Object]", but when I look at your code there should be all kinds of elements in this object; can I change something in your code to see the details of the object "evohome-session"?

**Comment by Meewis** [comment]
With a bit of debugging I just noticed at evohome-control.js, it never passes this part: " if (session && session.isValid && session.isValid())". There is something wrong with the session status ("evohome-session"). Although getting data from the honeywell thermomstat works fine...

**Comment by dehsgr** [comment]
Ok @Meewis so you could give it a try to delete context data from node-red ui. They would be re-generated. Maybe they got corrupted.

**Comment by Meewis** [comment]
Hi dehsgr, I did delete the context data and indeed it came back after a new status request, but with the same content; "[object Object]" And after the second request I also get a good data reply: {"id":"7646762","name":"thermostat","currentTemperature":19,"targetTemperature":11.5,"systemModeStatus":{"mode":"Auto","isPermanent":true}}

**Comment by dehsgr** [comment]
The context data should be ok now.

**Comment by Meewis** [comment]
It should be, but unfortunately it is not...

**Comment by dehsgr** [comment]
@Meewis I would like you to try latest changes. If you confirm they're working, I'll publish them to NPM.

**Comment by Meewis** [comment]
Good morning dhsgr, I will do that at the coming weekend. Thank you for your support. Kind regards Wilco

**Comment by Meewis** [comment]
Good afternoon dehsgr, unfortunately no improvements with the 1.3.0 version, nothing changed..

**Comment by Meewis** [comment]
Hello dehsgr, I did some debugging and adjustment to your code and now it is working. Besides these adjustments in the code, it is good to notice that the msg.payload to adjust the temperature, has to be an object.
I changed the evohome-control.js file for 2 parts; the var ' renew'  was not defined and the ' permanent = true'  does not work without a schedule. So I made jump over the schedule part when permanent=true.
Now it working and I'm happy.
[evohome-control.js.txt](https://github.com/dehsgr/node-red-contrib-evohome/files/9794141/evohome-control.js.txt)
